src/fix_nh.cpp

```diff
@@ -483,6 +505,13 @@ FixNH::FixNH(LAMMPS *lmp, int narg, char **arg) :
     if (p_flag[3]) box_change |= BOX_CHANGE_YZ;
     if (p_flag[4]) box_change |= BOX_CHANGE_XZ;
     if (p_flag[5]) box_change |= BOX_CHANGE_XY;
+    // In the isochoric case, dimensions can change size while not being
+    // barostated to maintain volume/surface
+    if (isochoric) {
+      if (not p_flag[0]) box_change |= BOX_CHANGE_X;
+      if (not p_flag[1]) box_change |= BOX_CHANGE_Y;
+      if (dimension == 3 && not p_flag[2]) box_change |= BOX_CHANGE_Z;
```

Comment:
```suggestion
      if (dimension == 3 && !p_flag[2]) box_change |= BOX_CHANGE_Z;
```

@Bibobu here is the Windows issue.
Using "not" instead of "!" is not portable to MS Visual C++.